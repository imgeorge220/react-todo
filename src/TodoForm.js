import React, { useState } from 'react';

function TodoForm({ addTodo }) {
  const INITIAL_STATE = { msg: "" };
  const [formData, setFormData] = useState(INITIAL_STATE);

  const handleSubmit = evt => {
    evt.preventDefault();
    addTodo(formData);
    setFormData(INITIAL_STATE);
  }

  const handleChange = evt => {
    const { name, value } = evt.target;
    setFormData(oldFormData => ({
      ...oldFormData,
      [name]: value
    }));
  }

  return (
    <div className="TodoForm">
      <form onSubmit={handleSubmit}>
        <label htmlFor="msg">Todo:</label>
        <input
          id='msg'
          name="msg"
          value={formData.msg}
          onChange={handleChange}
        />
        <button type='submit'>Add Todo!</button>
      </form>
    </div>
  );
}

export default TodoForm;
