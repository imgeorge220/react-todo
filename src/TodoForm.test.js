import React from "react";
import { render } from "@testing-library/react";
import TodoForm from "./TodoForm";

it("renders without crashing", function(){
  render(<TodoForm />);
});

it("matches snapshot", function(){
  const { asFragment } = render(<TodoForm />);
  expect(asFragment()).toMatchSnapshot();
});