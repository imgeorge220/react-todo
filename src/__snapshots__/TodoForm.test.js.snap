// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="TodoForm"
  >
    <form>
      <label
        for="msg"
      >
        Todo:
      </label>
      <input
        id="msg"
        name="msg"
        value=""
      />
      <button
        type="submit"
      >
        Add Todo!
      </button>
    </form>
  </div>
</DocumentFragment>
`;
